Tests/Unit/Service/ContentServiceTest.php

```diff
@@ -393,4 +478,92 @@ public function moveRecordWithPositiveRelativeToLoadsRelativeCopiesValuesSetsCol
+     * @param array $relativeToRecord (partial) record to be returned if queried for absolute $relativeTo value as content uid (return value for ContentService::loadRecordFromDatabase mock)
+     * @param array $targetAreaStoredInSession return value for ContentService::getTargetAreaStoredInSession mock
+     * @param array $parameters parameters as given to moveRecord
+     * @param int $expectedColPos the expected colPos after moveRecord has been called
+     */
+    public function moveRecordChangesColPos($rowBefore, $relativeTo, $relativeToRecord, $targetAreaStoredInSession, $parameters, $expectedColPos)
+    {
+        $mock = $this->getMock(
+            $this->createInstanceClassName(),
+            array('loadRecordFromDatabase', 'updateRecordInDatabase', 'updateMovePlaceholder', 'getTargetAreaStoredInSession')
+        );
+        $mock->expects($this->any())->method('loadRecordFromDatabase')->with(abs($relativeTo))->will($this->returnValue($relativeToRecord));
+        $mock->expects($this->any())->method('getTargetAreaStoredInSession')->with($relativeTo)->will($this->returnValue($targetAreaStoredInSession));
+        $mock->expects($this->any())->method('updateRecordInDatabase');
+        $mock->expects($this->any())->method('updateMovePlaceholder');
+        $dataHandler = $this->getMock('TYPO3\\CMS\\Core\\DataHandling\\DataHandler', array('resorting'));
+        $dataHandler->expects($this->any())->method('resorting');
+
+        $row = $rowBefore; // row will be modified by reference
+        $mock->moveRecord($row, $relativeTo, $parameters, $dataHandler);
+
+        $this->assertEquals($expectedColPos, $row['colPos']);
+    }
+
+    /**
+     * @return array
+     */
+    public function getmoveRecordChangesColPosTestValues()
+    {
+        $colPosFluxContent = 18181;
+
+        return [
+            // cases where colPos may change due to relative record using different colPos (starting with colPos 2, colPos 1 on relative record, neither flux parent or column name is set)
+            [['colPos' => 2], 0 - MiscellaneousUtility::UNIQUE_INTEGER_OVERHEAD - 1, [], [0, ''], ['', ''], 2], // using session
+            [['colPos' => 2], 123, ['tx_flux_column' => '', 'tx_flux_parent' => 0, 'colPos' => 1], [], ['', 'colpos-1-page-unused-unused-top-0-'], 1], // using parameters
+            [['colPos' => 2], -23, ['tx_flux_column' => '', 'tx_flux_parent' => 0, 'colPos' => 1], [], ['', ''], 1], // "inserting new element after"
+            [['colPos' => 2], 123, ['tx_flux_column' => '', 'tx_flux_parent' => 0, 'colPos' => 1], [], ['', ''], 2], // "first position in colPos"
+            [['colPos' => 2], 0, ['tx_flux_column' => '', 'tx_flux_parent' => 0, 'colPos' => 1], [0, ''], ['', ''], 2], // unmatched inner cases
+            [['colPos' => 2], 'x123', ['tx_flux_column' => '', 'tx_flux_parent' => 0, 'colPos' => 1], [], ['', ''], 2], // not changing for gridelements
```

Comment:
Please remove reference to gridelements (will show up in code searches) and rename the test set to something like `not changing on unrecognized colPos`. Small note: if you use the test set description as key, it will show up if/when the test fails with that specific test set - which would be very nice since you added all these gorgeous test cases!